Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/WorkspaceController.java

```diff
@@ -89,24 +94,50 @@ public String getWorkspaceDetails(@PathVariable("workspaceid") String workspacei
             ModelMap model) throws QuadrigaStorageException, QuadrigaAccessException, Quadriga404Exception {
         String userName = principal.getName();
         IWorkSpace workspace = wsManager.getWorkspaceDetails(workspaceid, userName);
+        String projectId = wsManager.getProjectIdFromWorkspaceId(workspaceid);
```

Comment:
if there is no workspace with this id, then this will result in null or an exception (and the next line too). So, you should check if the workspace is null first.